test/logger/MeetingSessionPOSTLogger.test.ts

```diff
@@ -0,0 +1,351 @@
+          'testLogger',
+          configuration,
+          batchSize,
+          intervalMs,
+          BASE_URL,
+          LogLevel.OFF
+        );
+        logger.info('info');
+        logger.error('error');
+        expect(logger.getLogCaptureSize()).is.equal(0);
+        logger.stop();
+        done();
+      });
+    });
+
+    it('should skip info and debug logs by default', done => {
+      new TimeoutScheduler(20).start(() => {
+        const logger: MeetingSessionPOSTLogger = new MeetingSessionPOSTLogger(
+          'testLogger',
+          configuration,
+          batchSize,
+          intervalMs,
+          BASE_URL
+        );
+        logger.debug(() => {
+          return 'debug';
+        });
+        logger.info('info');
+        logger.warn('warn');
+        logger.error('error');
+        expect(logger.getLogCaptureSize()).is.equal(2);
+        logger.stop();
+        done();
+      });
+    });
+
+    it('should have debug and info logs after setting DEBUG log level', done => {
+      new TimeoutScheduler(20).start(() => {
+        const logger: MeetingSessionPOSTLogger = new MeetingSessionPOSTLogger(
+          'testLogger',
+          configuration,
+          batchSize,
+          intervalMs,
+          BASE_URL
+        );
+        logger.debug(() => {
+          return 'debug';
+        });
+        expect(logger.getLogCaptureSize()).is.equal(0);
+        expect(logger.getLogLevel()).to.equal(LogLevel.WARN);
+        logger.setLogLevel(LogLevel.DEBUG);
+        expect(logger.getLogLevel()).to.equal(LogLevel.DEBUG);
+        logger.debug(() => {
+          return 'debug';
+        });
+        logger.info('info');
+        expect(logger.getLogCaptureSize()).is.equal(2);
+        logger.stop();
+        done();
+      });
+    });
+
+    it('should start publishing logs', done => {
+      new TimeoutScheduler(20).start(() => {
+        const logger: MeetingSessionPOSTLogger = new MeetingSessionPOSTLogger(
+          'testLogger',
+          configuration,
+          batchSize,
+          intervalMs,
+          BASE_URL
+        );
+        logger.startLogPublishScheduler(batchSize);
+        logger.stop();
+        done();
+      });
+    });
+  });
+
+  describe('startLogPublishSchedulerSpy', () => {
+    it('can call startLogPublishSchedulerSpy', done => {
+      new TimeoutScheduler(20).start(() => {
+        const logger: MeetingSessionPOSTLogger = new MeetingSessionPOSTLogger(
+          'testLogger',
+          configuration,
+          batchSize,
+          intervalMs,
+          BASE_URL
+        );
+        const startLogPublishSchedulerSpy = sinon.spy(logger, 'startLogPublishScheduler');
+        logger.startLogPublishScheduler(batchSize);
+        expect(startLogPublishSchedulerSpy.calledOnce).to.be.true;
+        logger.stop();
+        done();
+      });
+    });
+
+    it('handles when the fetch call fails', done => {
+      new TimeoutScheduler(20).start(() => {
```

Comment:
You can remove outer `TimeoutScheduler` from all of your tests but just need an inner one (line 219) for testing the interval scheduler.